scripts/generate_async_api.py

```diff
@@ -58,7 +58,21 @@ def generate(t: Any) -> None:
         print(f"{prefix}{suffix}")
     for [name, value] in t.__dict__.items():
         if name.startswith("_"):
-            continue
+            if name in ["__aenter__", "__aexit__"]:
```

Comment:
Instead of growing the generator, introduce `SyncBaseContextManager` and `AsyncBaseContextManager` that would implement enter/exit and only inherit Page, BrowserContext and Browser from them.

Reply:
Were you going to address this or should I close the PR? Either way is fine!

Reply:
I didn't understood how you wanted to do it, so closing